vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -416,6 +425,37 @@ private void unstashWorkspace(config, prefix) {
     }
 }
 
+private List getInitContainerList(config){
+    def initContainerSpecList = []
+    if (config.initContainerImage && config.containerMountPath) {
+        // regex [\W_] matches any non-word character equivalent to [^a-zA-Z0-9_]
+        def initContainerName = config.initContainerImage.replaceAll(/[\W_]/,"_" )
+        def initContainerSpec = [
+            name           : initContainerName,
+            image          : config.initContainerImage
+            ]
+        if (config.containerMountPath) {
+            initContainerSpec.volumeMounts = [[name: "volume", mountPath: config.containerMountPath]]
```

Comment:
Having these maps inside lists is a tricky business. Have you done some real world tests that confirm the pod spec turns out correctly?

Reply:
> Having these maps inside lists is a tricky business. Have you done some real world tests that confirm the pod spec turns out correctly?

https://gkecloudciservice.jaas-gcp.cloud.sap.corp/view/all/job/test_additionalpodproperty_step/job/step_test/75/console

Reply:
Yes parameters are already tested